frontend/catalyst/pennylane_extensions.py

```diff
@@ -465,6 +467,95 @@ def _check(x, hint):
     return jprim.vjp_p.bind(*params, *cotangents, jaxpr=jaxpr, fn=fn, grad_params=grad_params)
 
 
+class Adjoint(Operation):
+    """A minimal implementation of PennyLane operation, designed with a sole purpose of being
+    placed on the quantum tape"""
+
+    num_wires = AnyWires
+
+    def __init__(self, body_jaxpr, consts_jaxpr, cargs_jaxpr):
+        self.body_jaxpr = body_jaxpr
+        self.consts_jaxpr = list(consts_jaxpr)
+        self.cargs_jaxpr = list(cargs_jaxpr)
+        super().__init__(wires=Wires(Adjoint.num_wires))
+
+
+def adjoint(f: Union[Callable, Operator]) -> Union[Callable, Operator]:
+    """Getting the quantum adjoint of the computation defined by `f`.
+    Args:
+        f (Callable or Operator): A PennyLane operation or a Python function containing PennyLane
+          quantum operations. Quantum measurements are not allowed. The Catalyst control
+          flow instructions are not supported yet.
+    Returns:
+        If `f` is a callable:
+            A callable object accepting the same parameters as `f` but performing the computation
+            adjointed to what `f` does.
+        Otherwize, assuming that `f` is an operator:
+            The operator which is adjointed to `f`
```

Comment:
```suggestion
    Args:
        f (Callable or Operator): A PennyLane operation or a Python
        function containing PennyLane quantum operations.

    Returns:
        If an Operator is provided, returns an Operator that is the adjoint. If a function is provided, returns a function with the same call signature that returns the Adjoint of the provided function.
```

@grwlf slight modification to bring the docstring in line with https://docs.pennylane.ai/en/stable/code/api/pennylane.adjoint.html :)